src/main/java/io/reactivex/Completable.java

```diff
@@ -394,6 +394,17 @@ public static Completable fromRunnable(final Runnable run) {
     /**
      * Returns a Completable instance that subscribes to the given publisher, ignores all values and
      * emits only the terminal event.
+     * <p>
+     * The {@link Publisher} must follow the
+     * <a href="https://github.com/reactive-streams/reactive-streams-jvm#reactive-streams">Reactive-Streams specification</a>.
+     * Violating the specification may result in undefined behavior.
+     * <p>
+     * If possible, use {@link #create(CompletableOnSubscribe)} to create a
+     * source-like {@code Completable} instead.
+     * <p>
+     * Note that even though {@link Publisher} appears to be a functional interface, it
+     * is not recommended to implement it through a lambda as the specification requires
+     * state management not achievable with a stateless lambda.
```

Comment:
maybe add `which is` here? state management `which is` not achievable with a stateless lambda.

Reply:
Or `that is`. Grammarly doesn't complain about the PR's text.

Reply:
That is would also work.